launch/world_modeling/image_recognition.launch

```diff
@@ -0,0 +1,71 @@
+
+    <arg name="color_extractor_srv_name" default="/$(arg robot_name)/people_recognition/extract_color" />
+    <arg name="openface_srv_name" default="face_recognition/recognize" />
+    <arg name="openpose_srv_name" default="/$(arg robot_name)/openpose/recognize" />
+    <arg name="keras_srv_name" default="face_recognition/get_face_properties" />
+
+    <arg name="openface_db" default="" />
+    <arg name="openface_config" default="$(env ROBOT_BRINGUP_PATH)/parameters/world_modeling/face_recognition.yaml" />
+    <arg name="keras_face_model" default="~/data/keras_models/age_gender/weights.28-3.73.hdf5" />
+
+    <arg name="tensorflow_graph" default="~/MEGA/data/$(arg robot_env)/models/image_recognition_tensorflow/output_graph.pb" />
+    <arg name="tensorflow_labels" default="~/MEGA/data/$(arg robot_env)/models/image_recognition_tensorflow/output_labels.txt" />
+
+    <!-- People recognition -->
+    <group ns="people_recognition">
+
+        <!-- People recognition 3D-->
+        <node name="people_recognition_3d_node" pkg="people_recognition_3d" type="people_recognition_3d_node" output="log" machine="$(arg machine)" />
+
+        <!-- People recognition 2D-->
+        <node name="people_recognition_2d_node" pkg="people_recognition_2d" type="people_recognition_2d_node" output="log" machine="$(arg machine)" >
+            <param name="color_extractor_srv_name" value="$(arg color_extractor_srv_name)" />
+            <param name="openface_srv_name" value="$(arg openface_srv_name)" />
+            <param name="openpose_srv_name" value="$(arg openpose_srv_name)" />
+            <param name="keras_srv_name" value="$(arg keras_srv_name)" />
+        </node>
+
+        <!-- Face recognition -->
+        <group ns="face_recognition" >
+            <!-- OpenFace detection -->
+            <node name="face_recognition" pkg="image_recognition_openface" type="face_recognition_node" output="log" machine="$(arg machine)">
+                <rosparam command="load" file="$(arg openface_config)" />
+                <param name="topic_save_images" value="false" />
+                <param name="service_save_images" value="$(arg robot_real)" />
+                <param name="save_images_folder" value="~/MEGA/data/$(arg robot_env)/training_data" />
+                <param name="db" value="$(arg openface_db)" />
+            </node>
+
+            <!-- Keras face properties -->
+            <node pkg="image_recognition_keras" type="face_properties_node" name="keras_face_properties">
+                <param name="weights_file_path" value="$(arg keras_face_model)" />
+                <param name="save_images" value="$(arg robot_real)" />
+                <param name="save_images_folder" value="~/MEGA/data/$(arg robot_env)/training_data" />
+            </node>
+        </group>
+
+        <!-- Colour extraction -->
+        <node pkg="image_recognition_util" type="color_extractor_node" name="color_extractor_node" />
+    </group>
+
+    <!-- Object recognition -->
```

Comment:
Is there a need to launch object recognition and people recognition in the same launch file? Because sometimes we may need one but not the other.

Reply:
People recognition parts are just services so launching them wouldn't cause any effect on processing unless a service call is made. However, I agree with you that they probably never operate at the same time.

Reply:
never, that is very strong. They are just independent, so it is better to have separate launch files.

Reply:
Fixed

Also review and merge: https://github.com/tue-robotics/amigo_bringup/pull/107 and https://github.com/tue-robotics/sergio_bringup/pull/26